python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -138,9 +148,22 @@ def handle_output(self, context: OutputContext, obj: object) -> None:
 
     def load_input(self, context: InputContext) -> object:
         obj_type = context.dagster_type.typing_type
-        self._check_supported_type(obj_type)
+        if obj_type is Any and self._default_load_type is not None:
+            msg = (
+                f"Input {context.name} does not have a type hint, loading as default type"
```

Comment:
I imagine this warning could get kind of annoying. Do we need it?

Reply:
To expand on that a little bit, I think that in some situations, this will be a perfectly reasonable way to use these IO managers, and a logging a warning makes it seem like we're trying to discourage it.

Reply:
hmm ben had a comment about indicating that we were loading in this way, so i experimented with this. but i dont see the comment here anymore. maybe mentioning the behavior in docs would be enough?

Reply:
😅 I ended up thinking over the use-case where the default was applied and felt less strongly about my comment, so I deleted it. Sorry about any confusion! I think documenting the behavior might be sufficient, since users importing and using e.g. the preconstructed `snowflake_pandas_io_manager` would hit this warning otherwise which feels odd.

Reply:
ah ok that makes a lot more sense now. I'm happy with a docs change!

Reply:
API docs have been updated. I'm not sure it's worth a callout in the guide, but happy to be convinced otherwise